### DISCUSSION TITLE
Markor snippet / template syntax

### DISCUSSION THREAD

**Discussion by harshad1** [gsantner/markor#2221] [General]
This should be added to the README, but for now I am adding this information here. The following strings will be interpolated / substituted in snippets / templates when they are inserted:

- {{sel}}: replaced with currently selected text
- {{date}}: Inserts today's date in yyyy-MM-dd format
- {{time}}: Inserts the current time in HH:mm format
- {{title}}: Inserts current / new document title
- {{cursor}}: Cursor placed here after snippet insertion
- {{uuid}}: Inserts a uuid4

Any text within backtics `` `yyyy-MM-dd` `` will be interpreted as a java simpledateformat.

**Comment by ghost** [comment]
But how can this be used to create the default title for a new note?

**Reply by gsantner** [reply]
Not. It is for content.

**Reply by Toremok** [reply]
Hi, great application!

Does that mean there is no way to autofill the date in the default title for a new note? I noticed in the template examples, "2029-01-01-jekyll-post.md" (as well as one of the zettlkasten) automatically adds the date to the title. How would I add that feature to a custom template? Is it possible? 

Also, are the example templates hidden in /snippets? I can't seem to find them. 
Can I remove them from the list without deleting the files themselves?

Thank you very much.

**Reply by ghost** [reply]
There's an approved issue for this: https://github.com/gsantner/markor/issues/2225

**Comment by harshad1** [comment]
The very latest builds include some templating ability in the new file dialog

![Screenshot_20240605-130726](https://github.com/gsantner/markor/assets/1940940/0f5f5df9-c1e8-4fcb-9fb9-ce92a680c2fd)

**Reply by sergifornosgonzalez** [reply]
How do you delete a custom template? I created one as a test and it keeps showing up. I didn't find anything on the settings menu.

![Screenshot_2025-08-20-19-32-44-69_7aabe639e29ea39a625bdcb66fe7cfab](https://github.com/user-attachments/assets/bbb7fa07-eb0c-4f60-8268-1c8a3703d04e)

**Reply by Nalci** [reply]
I have the same question. I experimented with different formats and now they clutter the format list. I could not find an option to delete them.

**Comment by hmmuaa** [comment]
how to use `selection`?

my snippet file: `<ruby>{{selection}}{{cursor}}<rt>`

when using it I got `<ruby>{{selection}}<rt>`
(the `cursor` works correctly)

**Reply by f-Forneus** [reply]
do you have to use ruby? could I HTML?

**Comment by harshad1** [comment]
{{sel}} works.

I will update

**Comment by harshad1** [comment]
Also note that each template format will remember the title format used.

**Comment by clintonthegeek** [comment]
What string will allow me to choose where shared text will go within a template?

I'd like to use the Android Share feature to send text to a new note with a custom template which requires boilerplate at the ending, so the text has to go somewhere in the middle of the template doc, not be appended to the end.

Thanks!

**Comment by harshad1** [comment]
![Screenshot_20240806-074307](https://github.com/user-attachments/assets/02710b30-114f-4ac1-a2ab-3ece52afb144)

**Reply by nutpantz** [reply]
i tried adding {{text}} to a template, but the text i share into is always appended to the end.  can you provide an example?

**Comment by clintonthegeek** [comment]
Thank you, I had noticed that option. This would format every text shared-into Markor, would it not? All I'd like is for this trailing boilerplate to be part of one template only.

Specifically, I'd like to get an encryption/decryption chain working between OpenKeychain and Markor with QOwnNotes compatible markdown.

That is, the template needs to look something like this:

````
 {{title}}
 ===========

 <!-- BEGIN ENCRYPTED TEXT --
{{sharedintotext}}
  -- END ENCRYPTED TEXT -->
````

Where `{{sharedintotext}}` will be a PGP block. This way, these notes will be seamlessly encrypted and decrypted in QOwnNotes on my desktop, and can also be seamless done so in OpenKeychain.

If I use the option you've suggested, then the phrase ``  -- END ENCRYPTED TEXT -->`` would end up being appended to *everything* I share into Markor, no?

**Reply by nutpantz** [reply]
i am also looking for some way to accomplish this

preferably with selected text or text shared into markor

**Comment by cronyakatsuki** [comment]
Any way to get calculate dates?

As in I want to be able to get yesterday and tomorrow date and link them in my markdown file.

**Reply by harshad1** [reply]
No. I want this too. Investigated full js interpolation before abandoning it.

Need a good syntax, ideally one compatible with other apps

**Comment by gsantner** [comment]
Ruby was just part of the text, which could have been -hello- as well. It doesn't relate to ruby syntax / ruby programming in this case.

**Comment by gsantner** [comment]
there is a .app subfolder in your main directory.

If its not visible, top bar - dotfiles.

**Comment by MRDGH2821** [comment]
Are there any additional time formats?
like `13:49:51`, `1:49 pm`, `01:49:51 pm` or [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601)

ISO 8601 isn't a valid file naming format, so it willl be usable only as text snippet inside a note. Can use `{{iso_time}}`

Timezone would also be nice - `{{tz}}` giving `+11:00` or `-04:30`

**Reply by gsantner** [reply]
The options available from simpledateformat (see also screenshot above from harshad1) should make it possible to customize it to the finest detail of your needs.

**Reply by MRDGH2821** [reply]
Thank you very much!
I just had to look up more about `simpledateformat`

**Comment by nutpantz** [comment]
just want to clarify 
{{sel}}: replaced with currently selected text

only works for text that is currenty selected in markor?

ie i cannot select text in my book reader or web browser then switch to markor and insert a snippt or create a new note with a template and have the text placed.

**Reply by harshad1** [reply]
Yeah it is currently selected text in markor

But we do have a format for the shared into text (settings -> other)

Expected use is inserting a timestamp with the text or something basic like that but you could potentially use it for something like this

Alternatively you could

1. share text into markor
2. select it
3. insert a snippet with {{sel}} int it
